pkg/controller/storagecluster/storagecluster.go

```diff
@@ -334,6 +339,78 @@ func (c *Controller) validateCustomAnnotations(current *corev1.StorageCluster) e
 	return nil
 }
 
+
+
+func getKeysFromNodeSelector(ns *corev1.NodeSelector) map[string]bool {
+	keys := make(map[string]bool)
+
+	// Use node name as selector.
+	if ns.NodeName != "" {
+		keys["NodeName"] = true
+	}
+
+	if ns.LabelSelector != nil {
+		for k, _ := range ns.LabelSelector.MatchLabels {
+			keys[k] = true
+		}
+
+		for _, r := range ns.LabelSelector.MatchExpressions {
+			keys[r.Key] = true
+		}
+	}
+
+	return keys
+}
+
+func (c *Controller) validateCloudStorageLabelKey(cluster *corev1.StorageCluster) error {
+	key, err := c.getCloudStorageLabelKey(cluster)
+	if err != nil {
+		return err
+	}
+
+	if cluster.Spec.CloudStorage != nil &&
+		cluster.Spec.CloudStorage.NodePoolLabel != "" &&
+		key != "" &&
+		cluster.Spec.CloudStorage.NodePoolLabel != key {
+		return fmt.Errorf("node pool label key incorrect, expected %s, actual %s", cluster.Spec.CloudStorage.NodePoolLabel, key)
```

Comment:
Shouldn't `key` be the expected and `Spec.CloudStorage.NodePoolLabel` be the actual?

Reply:
if nodePoolLabel is configured, then it should be used and expected. well depends on how we understand it, either one can be source of truth

Reply:
hmm.. that is true
Still I feel we should call `key` as expected so it tells the user that the `cloudStorage.NodePoolLabel` is the user defined key and it should match the expected value in `key`. Users can only modify that and not the 'expected'. Wdyt?